valhalla/midgard/encoded.h

```diff
@@ -177,7 +177,7 @@ std::string encode(const container_t& points, const int precision = ENCODE_PRECI
   // handy lambda to turn an integer into an encoded string
   auto serialize = [&output](int number) {
     // move the bits left 1 position and flip all the bits if it was a negative number
-    number = number < 0 ? ~(static_cast<unsigned int>(number) << 1) : (number << 1);
+    number = number < 0 ? ~(*reinterpret_cast<unsigned int*>(&number) << 1) : (number << 1);
```

Comment:
yeah so we want to tell the compiler hey just think of this signed integer as some bytes that we want to do bitwise manipulation to, that is what `reinterpret_cast` does for us. `static_cast` is meant to do an actual conversion which in this case we dont actually want and i just threw in there without thinking about it all those years ago